File: semester 4/MPP/JFTS/FestivalDomain/src/main/java/ro/mpp/domain/Artist.java
package ro.mpp.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Table;
import lombok.*;
import jakarta.persistence.Entity;

import java.sql.ResultSet;
import java.sql.SQLException;

@Getter
@ToString
@Entity
@Table(name = "artist")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Artist extends HasId<Integer> {
    @Column(name = "name")
    private String name;

    public Artist(ResultSet rs) throws SQLException {
        super(rs.getInt("id"));
        this.name = rs.getString("name");
    }

    public Artist(int id, String name) {
        super(id);
        this.name = name;
    }
}

File: semester 4/MPP/C#FTS/Model/Domain/Artist.cs
using System;
using System.Data;

namespace C_FTS.Domain
{
    public class Artist : Entity<int>
    {
        public string Name { get; set; }

        public Artist(int id,  string name) : base(id) { Name = name; }

        public Artist(IDataReader reader) : base(reader.GetInt32(reader.GetOrdinal("id")))
        {
            Name = reader.GetString(reader.GetOrdinal("name"));
        }

        public override string ToString()
        {
            return $"Artist(Id={Id}, Name={Name})";
        }
    }
}
